File: Client/src/app/features/vendors/data-access/models/get-vendor-by-id-response.ts
import { Address } from "../../../../shared/models/address-model";
import { ContactResponse } from "../../../../shared/models/contacts-response";

export interface GetVendorByIdResponse {
    readonly id: string;
    readonly name: string;
    readonly address: Address;
    readonly contacts: readonly ContactResponse[];
    readonly webpage: string | null;
}

File: AssetManagement.Application/Vendors/GetVendorById/GetVendorByIdResponse.cs

using AssetManagement.Application.Common.Models;

namespace AssetManagement.Application.Vendors.GetVendorById
{
    public sealed record class GetVendorByIdResponse(
        Guid Id,
        string Name,
        AddressResponse Address,
        string? Webpage,
        IReadOnlyCollection<ContactResponse> Contacts
        );
}
